modules/internal/src/main/scala/TaskImplicits.scala

```diff
@@ -15,25 +15,18 @@
  */
 
 package freestyle.rpc
-package async
+package internal
 
-import cats.~>
-import cats.arrow.FunctionK
-import cats.effect.IO
+import cats.effect.LiftIO
+import freestyle.rpc.internal.LiftTask._
 import monix.eval.Task
 import monix.execution.Scheduler
 
-import scala.concurrent.Future
+trait TaskImplicits {
 
-trait RPCAsyncImplicits {
+  implicit class TaskOps[A](task: Task[A]) {
 
-  implicit val future2Task: Future ~> Task = λ[Future ~> Task](Task.deferFuture(_))
+    def to[F[_]](implicit F: LiftIO[F], s: Scheduler): F[A] = effectLiftTask[F].liftTask(task)
```

Comment:
I would probably take `F: LiftTask[F]` here instead of `LiftIO` and `Scheduler`.

That would probably mean some changes in the macro as well.